spec/models/miq_worker/runner_spec.rb

```diff
@@ -69,4 +69,22 @@
       @worker_base.send(:process_message, "sync_config")
     end
   end
+
+  context "#initialize" do
+    let(:worker) do
+      server_id = EvmSpecHelper.local_miq_server.id
+      FactoryBot.create(:miq_worker, :miq_server_id => server_id, :type => "MiqGenericWorker")
+    end
+
+    let!(:runner) { MiqWorker::Runner.new(:guid => worker.guid) }
+
+    it "configures the #worker attribute correctly" do
+      expect(runner.worker.id).to eq(worker.id)
+      expect(runner.worker.guid).to eq(worker.guid)
+    end
+
+    it "sets the MiqWorker.my_guid class attribute" do
+      expect(MiqWorker.my_guid).to eq(worker.guid)
+    end
```

Comment:
Does this need an "after" hook to clear the my_guid value?  Or should it live in EvmSpecHelper.clear_caches?

Reply:
Yeah, I think spec helper would be good for this.